crates/sncast/src/lib.rs

```diff
@@ -252,80 +275,104 @@ pub async fn check_class_hash_exists(
+        get_from_json(entry, parent_key, child_key)
             .and_then(Value::as_str)
-            .ok_or_else(|| anyhow::anyhow!("Failed to get address from account JSON file - make sure the account is deployed"))?
-    )?;
+            .map(str::to_string)
+    };
 
-    Ok(SingleOwnerAccount::new(
-        provider,
-        signer,
+    let address = parse_to_string(&account_info, "deployment", "address").ok_or_else(|| {
+        anyhow::anyhow!(
+            "Failed to get address from account JSON file - make sure the account is deployed"
+        )
+    })?;
+    let public_key = parse_to_string(&account_info, "variant", "public_key")
+        .ok_or_else(|| anyhow::anyhow!("Failed to get public key from account JSON file"))?;
+
+    let class_hash = parse_to_string(&account_info, "deployment", "class_hash");
+    let salt = parse_to_string(&account_info, "deployment", "salt");
+    let deployed =
+        parse_to_string(&account_info, "deployment", "status").map(|status| &status == "deployed");
+    let legacy = get_from_json(&account_info, "variant", "legacy").and_then(Value::as_bool);
+
+    Ok(AccountData {
+        private_key,
+        public_key,
         address,
-        chain_id,
-        ExecutionEncoding::Legacy,
-    ))
+        salt,
+        deployed,
+        class_hash,
+        legacy,
+    })
 }
 
-fn get_account_from_accounts_file<'a>(
+fn get_from_json<'a>(entry: &'a Value, parent_key: &str, child_key: &str) -> Option<&'a Value> {
+    entry
+        .get(parent_key)
+        .expect("Failed to get key from the account JSON file")
+        .get(child_key)
+}
+
+fn get_account_data_from_accounts_file(
     name: &str,
-    accounts_file_path: &Utf8PathBuf,
-    provider: &'a JsonRpcClient<HttpTransport>,
     chain_id: FieldElement,
-) -> Result<SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>> {
-    account_file_exists(accounts_file_path)?;
-    let account_info = get_account_info(name, chain_id, accounts_file_path)?;
-    let signer = LocalWallet::from(SigningKey::from_secret_scalar(
-        FieldElement::from_hex_be(&account_info.private_key).with_context(|| {
-            format!(
-                "Failed to convert private key = {} to FieldElement",
-                &account_info.private_key
-            )
-        })?,
-    ));
-    let address = FieldElement::from_hex_be(&account_info.address).with_context(|| {
-        format!(
-            "Failed to convert account address = {} to FieldElement",
-            &account_info.address
+    path: &Utf8PathBuf,
+) -> Result<AccountData> {
+    raise_if_empty(name, "Account name")?;
+    check_account_file_exists(path)?;
+
+    let accounts: HashMap<String, HashMap<String, AccountData>> = read_and_parse_json_file(path)?;
+    let network_name = chain_id_to_network_name(chain_id);
+
+    let account = accounts
+        .get(&network_name)
+        .and_then(|accounts_map| accounts_map.get(name))
+        .cloned();
```

Comment:
it would be cool to get rid of this clone, we could probably just merge this + ok_or_else (below) and return directly

Reply:
I don't see an easy way to get rid of `cloned` at the moment, but I simplified the code slightly in https://github.com/foundry-rs/starknet-foundry/pull/1850/commits/a5edc902094e1d45f6f0c68ee4db5e9becadb6d3